packages/admin/src/common/toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,27 @@
+import { Paper, Toolbar as MUIToolbar } from "@material-ui/core";
+import * as React from "react";
+
+import { MenuContext } from "../../mui";
+import { IStackSwitchApi, useStackSwitchApi } from "../../stack";
+import { useStyles, useThemeProps } from "./Toolbar.styles";
+
+interface ToolbarProps {
+    actionItems?: (stackSwitchApi: IStackSwitchApi | undefined) => React.ReactNode;
+}
+const Toolbar: React.FunctionComponent<ToolbarProps> = ({ actionItems, children }) => {
+    const themeProps = useThemeProps();
+    const { headerHeight } = React.useContext(MenuContext);
+    const classes = useStyles({ headerHeight });
+    const stackSwitchApiContext = useStackSwitchApi();
+
+    return (
+        <Paper elevation={themeProps.elevation} className={classes.root}>
+            <MUIToolbar classes={{ root: classes.muiToolbar }}>
+                <div className={classes.mainContentContainer}>{children}</div>
+                <div className={classes.actionContainer}>{actionItems?.(stackSwitchApiContext)}</div>
```

Comment:
Working with JSX-Elements  (JSX-Elements call the react-top-level api "React.createElement" under the hood), should be preferred IMO over functions that return JSX (non-react-api).

```
{ActionItems && <ActionItems stackSwitchApi={stackSwitchApi} /> }

```

instead of

```
actionItems?.(stackSwitchApiContext)
```

Reply:
I wouldn't inject stackSwitchApi at all. It's not the responsibility of the Toolbar to inject random (although common) dependencies.

Reply:
@nsams thats definitly true. will remove that, thanks for pointing this out.